issue #8 - add support for Basic auth

jmazzitelli opened the pull request.

burmanm commented on metrics/client.go:
```diff
@@ -592,16 +597,22 @@ func NewHawkularClient(p Parameters) (*Client, error) {
 	}
 
+	var creds string
+	if p.Username != "" && p.Password != "" {
```
Only token or or only username/password can be allowed. Otherwise the client should throw an error when trying to initialize.

Having both would mean that the token overwrites the username/password authorization header. So this should check that the token isn't also set (or the token set should check that)

jmazzitelli commented:
NewHawkularClient no longer allows for both Username/Password and Token to be supplied. You must specify none or just one of them, but not both.

Tests added to test that you can only specify one of Username/Password or Token. Test also added to make sure both Username and Password are there (you can't have Username without Password and you can't have Password without Username).

Test is also added to make sure the Basic header is set correctly.

jmazzitelli commented:
FWIW: I can confirm this new feature works. I'm using my branch in my Go agent - the client can successfully write data to H-Metrics and its using this new Basic auth.

burmanm commented on metrics/client_test.go:
```diff
@@ -290,6 +291,72 @@ func TestTokenAuthenticationWithSSL(t *testing.T) {
+	}
+
+	_, err = NewHawkularClient(p)
+	if err == nil {
```
`assert.Error(t, err)` would be more consistent, since everything else uses an error. Or even `assert.ErrorEquals` if you want to check the returned error is correct.

burmanm commented on metrics/client_test.go:
```diff
@@ -290,6 +291,72 @@ func TestTokenAuthenticationWithSSL(t *testing.T) {
+	}
+
+	_, err = NewHawkularClient(p)
+	if err == nil {
```
Same here, use assert

jmazzitelli commented:
Now uses assert.Error.
Commits squashed (though, note that github now has the ability to squash merge - see drop down in the green "Merge pull request" button - so the squash can now happen at PR merge time).
